ironfish-cli/src/commands/service/faucet.ts

```diff
@@ -143,38 +147,44 @@ export default class Faucet extends IronfishCommand {
 
     this.warnedFund = false
 
-    const faucetTransaction = await api.getNextFaucetTransaction()
+    const faucetTransactions = await api.getNextFaucetTransactions(
+      MAX_RECIPIENTS_PER_TRANSACTION,
+    )
 
-    if (!faucetTransaction) {
+    if (!faucetTransactions) {
       this.log('No faucet jobs, waiting 5s')
       await PromiseUtils.sleep(5000)
       return
     }
 
-    this.log(`Starting ${JSON.stringify(faucetTransaction, undefined, '   ')}`)
+    this.log(`Starting ${JSON.stringify(faucetTransactions, undefined, '   ')}`)
+
+    for (const faucetTransaction of faucetTransactions) {
+      await api.startFaucetTransaction(faucetTransaction.id)
+    }
 
-    await api.startFaucetTransaction(faucetTransaction.id)
+    const receives = faucetTransactions.map((ft) => {
+      return {
+        publicAddress: ft.public_key,
+        amount: BigInt(FAUCET_AMOUNT).toString(),
+        memo: `Faucet for ${ft.id}`,
+      }
+    })
 
     const tx = await client.sendTransaction({
       fromAccountName: account,
-      receives: [
-        {
-          publicAddress: faucetTransaction.public_key,
-          amount: BigInt(FAUCET_AMOUNT).toString(),
-          memo: `Faucet for ${faucetTransaction.id}`,
-        },
-      ],
+      receives,
       fee: BigInt(FAUCET_FEE).toString(),
     })
 
     speed.add(1)
 
-    this.log(
-      `COMPLETING: ${faucetTransaction.id} ${tx.content.hash} (5m avg ${speed.rate5m.toFixed(
-        2,
-      )})`,
-    )
+    const ids = faucetTransactions.map((ft) => ft.id).join(', ')
 
-    await api.completeFaucetTransaction(faucetTransaction.id, tx.content.hash)
+    this.log(`COMPLETING: ${ids} ${tx.content.hash} (5m avg ${speed.rate5m.toFixed(2)})`)
```

Comment:
Is this message easier to read this way? Do you prefer this over just doing what we did before?

Reply:
I don't prefer it actually; I intended on having a discussion on the topic, which is what we're in now! 👍🏽